File: output/scripts/mistsh/node/expr/number.lua
local modname = ...
local node = require('mistsh.node')
local numberexpr = node:module(modname)
local common

function numberexpr:init(pr, context)
	node.init(self, pr, context)
	self.value = pr._value
end

function numberexpr:defstring(lp)
	return string.format('%s', self.value)
end

function numberexpr:build()
	return string.format('%s', self.value)
end

common = require('mistsh.common')

File: exl/node/expr/number.cpp
#include <exl/node/expr/number.hpp>
#include <exl/value/number.hpp>
#include <exl/parser/ast.hpp>
#include <exl/construct.hpp>
#include <exl/format.hpp>

namespace exl
{
	using utils::SExpr;

	NumberExpr::NumberExpr( utils::SExpr const& s )
		: Expression( s )
	{
		ASSERT( s.head == NodeHead::expr_number );
		ASSERT( s.list.size() == 2 );
		ASSERT( s[ 2 ].head == SExpr::head_number );
		m_content = s[ 2 ].number;
	}

	NumberExpr::~NumberExpr()
	{
	}

	StringBuilder NumberExpr::getdefstring( size_t depth )
	{
		StringBuilder defstring;
		defstring << numtostr( m_content );
		return defstring;
	}

	void NumberExpr::build( IContext* context )
	{
		m_value = Ref< NumberValue >::create( m_content );
	}
}
